analyses/infercnv-consensus-cell-type/scripts/01b_run-infercnv-internal-reference.R

```diff
@@ -0,0 +1,305 @@
+    opt_str = c("--gene_order_file"),
+    type = "character",
+    default = file.path(project_root, "references", "Homo_sapiens.GRCh38.104.gene_order.txt"),
+    help = "Path to gene order file as tab delimited .txt file with no column headers.
+      Columns are: Ensembl gene id, chr, start, stop."
+  ),
+  make_option(
+    opt_str = "--annotation_file",
+    type = "character",
+    default = file.path(project_root, "scratch", "infercnv_annotation.txt"),
+    help = "Path to annotation file for inferCNV input"
+  ),
+  make_option(
+    opt_str = c("--scratch_dir"),
+    type = "character",
+    default = file.path(project_root, "scratch", "infercnv"),
+    help = "path to scratch directory to store intermediate files from infercnv"
+  ),
+  make_option(
+    opt_str = c("-t", "--threads"),
+    type = "integer",
+    default = 4,
+    help = "Number of multiprocessing threads to use."
+  ),
+  make_option(
+    opt_str = c("--seed"),
+    type = "integer",
+    default = 2025,
+    help = "Random seed to set for reproducibility. Note that inferCNV is only reproducible on a given operating system."
+  )
+)
+
+
+# Parse options ----------------
+opts <- parse_args(OptionParser(option_list = option_list))
+
+
+# check inputs
+stopifnot(
+  "sce_file does not exist" = file.exists(opts$sce_file),
+  "celltype_file does not exist" = file.exists(opts$celltype_file),
+  "reference_celltype_group not provided" = !is.null(opts$reference_celltype_group),
+  "gene_order_file does not exist" = file.exists(opts$gene_order_file),
+  "annotation_file not provided" = !is.null(opts$annotation_file),
+  "output_dir was not specified" = !is.null(opts$output_dir),
+  "hmm_model not properly specified" = opts$hmm_model %in% c("i3", "i6")
+)
+
+# Check the reference group
+celltype_groups <- c("endo", "immune", "endo-immune")
+stopifnot(
+  "The reference celltype group should be one of endo, immune, or endo-immune." =
+    opts$reference_celltype_group %in% celltype_groups
+)
+
+# make the resolution parameter numeric if it's not auto
+resolution <- opts$leiden_resolution
+if (resolution != "auto") {
+  resolution <- as.numeric(resolution)
+  stopifnot(
+    "The resolution parameter should either be a number or the string 'auto'." = !is.na(resolution)
+  )
+}
+
+# Setup files and paths -----------------
+set.seed(opts$seed)
+
+# read SCE file
+sce <- readRDS(opts$sce_file)
+
+# Read in cell type TSV
+celltype_df <- readr::read_tsv(opts$celltype_file)
+
+# Define cell groups
+immune_celltypes <- readr::read_tsv(opts$immune_ref_url) |>
+  dplyr::pull(consensus_annotation)
+endo_celltypes <- c(
+  "endothelial cell",
+  "blood vessel endothelial cell",
+  "microvascular endothelial cell"
+)
+if (opts$reference_celltype_group == "immune") {
+  reference_celltypes <- immune_celltypes
+} else if (opts$reference_celltype_group == "endo") {
+  reference_celltypes <- endo_celltypes
+} else {
+  reference_celltypes <- c(immune_celltypes, endo_celltypes)
+}
```

Comment:
Maybe we pull this out into its own little prep script that creates a tsv file with the assigned reference group and consensus annotation. Then that becomes an input to the build reference and run inferCNV script to match reference group names to consensus annotations?

Reply:
Yes, I'll do that separately!